bulkresetcompletion.php

```diff
@@ -0,0 +1,112 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Manually edit completion date for a course.
+ *
+ * @copyright 2020 Catalyst IT
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @package local_recompletion
+ */
+
+require_once('../../config.php');
+require_once($CFG->dirroot.'/user/lib.php');
+require_once($CFG->libdir.'/formslib.php');
+
+$courseid = required_param('id', PARAM_INT);
+$userid   = optional_param('user', 0, PARAM_INT);
+$users   = optional_param_array('users', array(), PARAM_INT);
+
+$course = $DB->get_record('course', array('id' => $courseid), '*', MUST_EXIST);
+require_login($course);
+
+$context = context_course::instance($course->id);
+require_capability('local/recompletion:manage', $context);
+
+$PAGE->set_url('/local/recompletion/editcompletion.php', array('id' => $course->id));
+if (empty($users) && empty($userid)) {
+    // The first time list hack.
+    if ($post = data_submitted()) {
+        foreach ($post as $k => $v) {
+            if (preg_match('/^user(\d+)$/', $k, $m)) {
+                $users[] = $m[1];
```

Comment:
I see this is old code copied from mine... but now that I look at that closely I don't like my code... :-)
can you please change this line to the following to cast the value from data_submitted as an int?
```
 $users[] = (int) $m[1];
```

Feel free to leave my old code on the other pages, I'll probably update that at the same time as merging this in :-)